src/Symfony/RequiredAutowiringExtension.php

```diff
@@ -0,0 +1,92 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Symfony;
+
+use PHPStan\Reflection\AdditionalConstructorsExtension;
+use PHPStan\Reflection\ClassReflection;
+use PHPStan\Reflection\Php\PhpPropertyReflection;
+use PHPStan\Reflection\PropertyReflection;
+use PHPStan\Rules\Properties\ReadWritePropertiesExtension;
+use PHPStan\Type\FileTypeMapper;
+use Symfony\Contracts\Service\Attribute\Required;
+use function count;
+
+class RequiredAutowiringExtension implements ReadWritePropertiesExtension, AdditionalConstructorsExtension
+{
+
+	/** @var FileTypeMapper */
+	private $fileTypeMapper;
+
+	public function __construct(FileTypeMapper $fileTypeMapper)
+	{
+		$this->fileTypeMapper = $fileTypeMapper;
+	}
+
+	public function isAlwaysRead(PropertyReflection $property, string $propertyName): bool
+	{
+		return false;
+	}
+
+	public function isAlwaysWritten(PropertyReflection $property, string $propertyName): bool
+	{
+		return false;
+	}
+
+	public function isInitialized(PropertyReflection $property, string $propertyName): bool
+	{
+		// If the property is public, check for @required on the property itself
+		if (!$property->isPublic()) {
+			return false;
+		}
+
+		if ($property->getDocComment() !== null && $this->isRequiredFromDocComment($property->getDocComment())) {
+			return true;
+		}
+
+		// Check for the attribute version
+		if ($property instanceof PhpPropertyReflection && count($property->getNativeReflection()->getAttributes(Required::class)) > 0) {
```

Comment:
`Class Symfony\Contracts\Service\Attribute\Required not found.` should be solved by doing `'Symfony\Contracts\Service\Attribute\Required'` as a literal string instead of `::class`.

As for the rest I'm gonna have to think about it and do something in PHPStan itself. Or maybe just ignore them, they're not that severe.

Reply:
Most issues are solved now, however using the class as a string, will either error out as it being a `string` and not a `class-string`, but when I explicitly type it as `class-string`, it errors with `PHPDoc tag @var with type class-string is not subtype of native type`. It's kind of fighting both sides this way. Is there perhaps a better solution? Stubs maybe?

Reply:
Okay, it took a little bit of effort figuring out the best way to approach this, but I've fixed it now!